mustache: fix missing support for partials for embedded filesystems

zachmann opened the pull request.

ReneWerner87 reviewed (changes requested):
thanks for the effort, can you add an example in the readme and write a unittest

zachmann commented:
I updated the PR:
- I updated the behavior of the `NewFileSystem` function; it will now use the passed filesystem also for partials. The other function I added `NewFileSystemPartials` can be used to set another filesystem for the partials.
  - **Note:** This changes the default behavior a bit (but it now should be what is expected, imho). The path of the included partials (specified in the template files) now must be relative to the embedded filesystem root and not to the current working directory as before.
- I updated the Readme; there already was a reference (in the comments) to using an embedded filesystem, I updated that a bit.
- Unittest: There also already was a unit test; I updated that to the (new) behavior. However, it is difficult to write a meaningful unittest for this, because the true meaning of the test is only given when executed from another working directory.
